### PR TITLE
fix: Use version instead of build number for notices

### PR DESCRIPTION
**Pull request by toninis**
#### Summary

Fix an erroneous log around build version. 

Related issue: https://github.com/mattermost/mattermost/issues/25335#issuecomment-1809405533
#### Release Note

```release-note

```

### CODE REVIEW

**Review by lieut-data — APPROVED** [review]

**Review by isacikgoz — APPROVED** [review]

**Review by amyblais — APPROVED** [review]

**Comment by isacikgoz** [comment]
Feel free to ask for another review once tests are updated @toninis

**Comment by saturninoabril** [comment]
Thanks @toninis! Again, I'll defer to @marianunez for this product change.

File: server/channels/app/product_notices.go
Comment (marianunez) [inline server/channels/app/product_notices.go:79]: @toninis Would this current version have any of the prefix or suffix that the cleanupVersion function was doing above? I believe the intent of that was it does a direct comparison with the versions that are included in the [notices.json](https://github.com/mattermost/notices/blob/master/notices.json) so it needs to clean up any additional prefix so it matches with the version strings in that notices definition

Comment (toninis) [reply server/channels/app/product_notices.go:79]: Current version should get the last value from https://github.com/mattermost/mattermost/blob/master/server/public/model/version.go#L15 always

Comment (marianunez) [reply server/channels/app/product_notices.go:79]: Change sounds good. However, the tests seem to be failing in cases where the version has additional info like hashes and this `cleanupVersion` would remove it. If we no longer support those cases then we need to remove the tests.

Comment (toninis) [reply server/channels/app/product_notices.go:79]: Yep . I need to fix the test 😄

**Review by marianunez — APPROVED** [review]
Changes LGTM - pending tests to be updated.

**Pull request merged**
